### *CHALLENGE*

##### You are a hospitality consultant that has been hired by Sunset Oasis Hotel in Las Vegas, Nevada. This hotel is still in the planning phase of its construction, giving you the ability to make actionable recommendations on its design. Your client has decided to position themselves to compete with the hotels in the Las Vegas dataset that has been provided, and they are very concerned about having high online reviews (especially from Tripadvisor where your dataset came from).
##### As construction has not started yet, Sunset Oasis Hotel would like your suggestions on how they can maximize their score on Tripadvisor. They have admitted that their statistical knowledge is not very advanced and have kindly requested that you explain your insights in a non-technical manner. They have also reminded you to support your insights with external research (web resources, research papers, etc.). They have also asked you to answer the following questions.
#####    •	Facilities: Given spatial limitations, should Sunset Oasis Hotel build a tennis court, a spa, or a casino? Based on your choice, how much of an increase should they expect in review scores on Tripadvisor?
#####    •	Targeting: What kind of travellers should Sunset Oasis Hotel target (businesses, couples, families, etc.)? Also, should they be targeting travellers in North America, or also from other parts of the world. How will this affect their review scores on Tripadvisor?
#####    •	Other: Based on the data, is there anything else Sunset Oasis Hotel should do that might improve their score on Tripadvisor?


### *EXPLORATORY DATA ANALYSIS*

##### Importing Libraries.

In [49]:
#IMport Statements
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns

##### Fetching data and checking the description of the data.

In [50]:
#Reading excel file.
data = pd.read_excel('vegas_data.xlsx')
#Checking first 5 rows of the dataset.
data.head()

Review ID User country  Nr. reviews  Nr. hotel reviews  Helpful votes  \
0          1          USA           11                  4             13   
1          2          USA          119                 21             75   
2          3          USA           36                  9             25   
3          4           UK           14                  7             14   
4          5       Canada            5                  5              2   

   Score Period of stay Traveler type Pool  Gym  ... Spa Casino Free internet  \
0      5        Dec-Feb       Friends   NO  YES  ...  NO    YES           YES   
1      3        Dec-Feb      Business   NO  YES  ...  NO    YES           YES   
2      5        Mar-May      Families   NO  YES  ...  NO    YES           YES   
3      4        Mar-May       Friends   NO  YES  ...  NO    YES           YES   
4      4        Mar-May          Solo   NO  YES  ...  NO    YES           YES   

                               Hotel name Hotel stars  Nr. rooms  \
0  Circus Circus Hotel & Casino Las Vegas         3.0       3773   
1  Circus Circus Hotel & Casino Las Vegas         3.0       3773   
2  Circus Circus Hotel & Casino Las Vegas         3.0       3773   
3  Circus Circus Hotel & Casino Las Vegas         3.0       3773   
4  Circus Circus Hotel & Casino Las Vegas         3.0       3773   

   User continent Member years  Review month Review weekday  
0   North America            9       January       Thursday  
1   North America            3       January         Friday  
2   North America            2      February       Saturday  
3          Europe            6      February         Friday  
4   North America            7         March        Tuesday  

[5 rows x 21 columns]

##### In the below code segment, info() method is used which clearly states the information of the attributes i.e. data type, null values, etc.

In [51]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 504 entries, 0 to 503
Data columns (total 21 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Review ID          504 non-null    int64  
 1   User country       504 non-null    object 
 2   Nr. reviews        504 non-null    int64  
 3   Nr. hotel reviews  504 non-null    int64  
 4   Helpful votes      504 non-null    int64  
 5   Score              504 non-null    int64  
 6   Period of stay     504 non-null    object 
 7   Traveler type      504 non-null    object 
 8   Pool               504 non-null    object 
 9   Gym                504 non-null    object 
 10  Tennis court       504 non-null    object 
 11  Spa                504 non-null    object 
 12  Casino             504 non-null    object 
 13  Free internet      504 non-null    object 
 14  Hotel name         504 non-null    object 
 15  Hotel stars        504 non-null    float64
 16  Nr. rooms          504 non

##### In the below code segment, describe() method is used which clearly states the statistical information of the attributes i.e. count, mean, standard deviation etc.

In [52]:
#Checking statistical description of data.
data.describe()

Review ID  Nr. reviews  Nr. hotel reviews  Helpful votes       Score  \
count  504.000000   504.000000         504.000000     504.000000  504.000000   
mean   252.500000    48.130952          16.069444      31.751984    4.123016   
std    145.636534    74.996426          23.928263      48.520783    1.007302   
min      1.000000     1.000000           1.000000       0.000000    1.000000   
25%    126.750000    12.000000           5.000000       8.000000    4.000000   
50%    252.500000    23.500000           9.000000      16.000000    4.000000   
75%    378.250000    54.250000          18.000000      35.000000    5.000000   
max    504.000000   775.000000         263.000000     365.000000    5.000000   

       Hotel stars    Nr. rooms  Member years  
count   504.000000   504.000000    504.000000  
mean      4.142857  2196.380952      4.359127  
std       0.774487  1285.476807      2.925856  
min       3.000000   188.000000      0.000000  
25%       3.500000   826.000000      2.000000  
50%       4.000000  2700.000000      4.000000  
75%       5.000000  3025.000000      6.000000  
max       5.000000  4027.000000     13.000000

##### Observing variables by making heatmap of Correlation. Correlation summarizes the strength and direction of the linear (straight-line) association between two quantitative variables. Denoted by r, it takes values between -1 and +1. A positive value for r indicates a positive association, and a negative value for r indicates a negative association.

In [75]:
#Pandas Built-in Correlation method.
data.corr()

Review ID  Nr. reviews  Nr. hotel reviews  Helpful votes  \
Review ID           1.000000     0.042557           0.043615       0.037251   
Nr. reviews         0.042557     1.000000           0.654819       0.790696   
Nr. hotel reviews   0.043615     0.654819           1.000000       0.764220   
Helpful votes       0.037251     0.790696           0.764220       1.000000   
Score               0.175538    -0.024320           0.012842       0.015472   
Hotel stars         0.055838    -0.048241          -0.090917      -0.019000   
Nr. rooms          -0.372522    -0.136580          -0.129739      -0.099148   
Member years        0.030907     0.117831           0.291504       0.269785   

                      Score  Hotel stars  Nr. rooms  Member years  
Review ID          0.175538     0.055838  -0.372522      0.030907  
Nr. reviews       -0.024320    -0.048241  -0.136580      0.117831  
Nr. hotel reviews  0.012842    -0.090917  -0.129739      0.291504  
Helpful votes      0.015472    -0.019000  -0.099148      0.269785  
Score              1.000000     0.191490  -0.080200      0.072673  
Hotel stars        0.191490     1.000000   0.318912      0.035657  
Nr. rooms         -0.080200     0.318912   1.000000     -0.017571  
Member years       0.072673     0.035657  -0.017571      1.000000

##### Checking if any data is missing (data cleaning)

In [54]:
#Determining if any value is missing in the dataframe.
data.isnull().values.any()

False

#### Q1a: Given spatial limitations, should Sunset Oasis Hotel build a tennis court, a spa, or a casino?

In [55]:
#Calculating average rating of all those hotels which have a "spa".
spa = data.query('Spa == "YES"')
average_rating_spa = round(spa["Hotel stars"].mean(),1)
average_rating_spa

4.3

##### The average rating of the hotels which have spa is 4.3.

In [56]:
#Calculating average rating of all those hotels which have a "tennis court".
tennis_court = data[data['Tennis court']=='YES']
average_rating_tennis_court = round(tennis_court["Hotel stars"].mean(),1)
average_rating_tennis_court

3.9

##### The average rating of the hotels which have tennis court is 3.9.

In [57]:
#Calculating average rating of all those hotels which have a "Casino".
casino = data.query('Casino == "YES"')
average_rating_casino = round(casino["Hotel stars"].mean(),1)
average_rating_casino

4.2

##### The average rating of the hotels which have casino is 4.2.

#### Q1b: Based on your choice, how much of an increase should they expect in review scores on Tripadvisor?

In [58]:
#Calculating average score of all those hotels which have a "spa".
average_score_spa = round(spa["Score"].mean(),1)

#Calculating average score of all those hotels which do not have a "spa".
without_spa = data.query('Spa == "NO"')
average_score_without_spa = round(without_spa["Score"].mean(),1)

#Calculating expected increase in review scores.
increase_in_score_spa = round((average_score_spa - average_score_without_spa), 1)
increase_in_score_spa

0.2

##### The increase in review scores will be 0.2, if the Sunset Oasis Hotel decides on building the spa in their hotel.

In [89]:
#Calculating average score of all those hotels which have a "tennis court".
average_score_tennis_court = round(tennis_court["Score"].mean(),1)

#Calculating average score of all those hotels which do not have a "tennis court".
without_tennis_court = data[data['Tennis court']=='NO']
average_score_without_tennis_court= round(without_tennis_court["Score"].mean(),1)

#Calculating expected increase in review scores.
increase_in_score_tennis_court = round((average_score_tennis_court - average_score_without_tennis_court), 1)
increase_in_score_tennis_court

0.1

##### The increase in review scores will be 0.1, if the Sunset Oasis Hotel decides on building the tennis court in their hotel. 

In [79]:
#Calculating average score of all those hotels which have a "Casino".
average_score_casino = round(casino["Score"].mean(),1)

#Calculating average score of all those hotels which do not have a "Casino".
without_casino = data.query('Casino == "NO"')
average_score_without_casino = round(without_casino["Score"].mean(),1)

#Calculating expected increase in review scores.
increase_in_score_spa = round((average_score_casino - average_score_without_casino), 1)
increase_in_score_spa

-0.1

##### The increase in review scores will be -0.1, if the Sunset Oasis Hotel decides on building the casino in their hotel. 

#### Q2a:  What kind of travellers should Sunset Oasis Hotel target (businesses, couples, families, etc.)?

##### In the below graph Traveler type is on y-axis and Score is on x-axis. The Barchart is created which is showing the relation between the mentioned attributes. 

In [80]:
#Grouping by Traveler type according to mean of Score.
traveler = data.groupby('Traveler type')['Score'].mean()
#Plotting the graph
fig1 = px.bar(data, y= traveler.index, x= traveler.values, labels={'x':'Score', 'y':'Traveler Type'}, orientation='h', color=traveler.index, title="Graph 1: Score by Traveler Type")
fig1.show()

##### In the below code segment, Total helpful votes are calculated according to the traveler types.

In [81]:
#Grouping by Traveler type according to sum of helpful votes.
data.groupby("Traveler type")["Helpful votes"].sum().sort_values(ascending=False)

Traveler type
Couples     6870
Business    3128
Families    2880
Friends     2416
Solo         709
Name: Helpful votes, dtype: int64

##### In the below graph Traveler type is on y-axis and Number of Travelers is on x-axis. The Barchart is created which is showing the relation between the mentioned attributes. 

In [82]:
#Grouping by Traveler type according to Number of travelers.
popular = data.groupby('Traveler type')['Traveler type'].count()
#Plotting the graph
fig2 = px.bar(data, y= popular.index, x= popular.values, labels={'y':'Traveler Type', 'x':'Number of Guests'}, orientation='h', color=popular.index, title = "Graph 2: Traveler Type by Number of Guests")
fig2.show()

#### Q2b:  Also, should they be targeting travellers in North America, or also from other parts of the world. How will this affect their review scores on Tripadvisor?

##### In the below graph Score is on y-axis and Countries is on x-axis. The Barchart is created which is showing the relation between the mentioned attributes.

In [83]:
#Grouping by User Country according to mean of Scores.
traveler_country = data.groupby('User country')['Score'].mean()
#Plotting the graph
fig3 = px.bar(data, y= traveler_country.values, x= traveler_country.index, labels={'y':'Score', 'x':'Countries'}, color_discrete_sequence=["sky blue"], title = "Graph 3: Score by Countries")
fig3.show()

##### In the below graph Number of Guests is on y-axis and Continents is on x-axis. The Barchart is created which is showing the relation between the mentioned attributes.

In [87]:
#Grouping by User Continent according to Number of Guests.
pop_continent = data.groupby('User continent')['User continent'].count()
#Plotting the graph
fig3 = px.bar(data, y= pop_continent.values, x= pop_continent.index, labels={'y':'Number of Guests', 'x':'Continents'}, color_discrete_sequence=["green"], title = "Graph 4: Number of Guests by Continents")
fig3.show()

#### Q3: Based on the data, is there anything else Sunset Oasis Hotel should do that might improve their score on Tripadvisor?

##### In the below graph Number of Guests is on y-axis and Countries is on x-axis. The Barchart is created which is showing the relation between the mentioned attributes.

In [90]:
#Grouping by User Country according to Number of Guests.
pop_country = data.groupby('User country')['User country'].count()
#plotting the graph
fig3 = px.bar(data, y= pop_country.values, x= pop_country.index, labels={'y':'Number of Guests', 'x':'Countries'}, color_discrete_sequence=["red"], title = "Graph 5: Number of Guests by Countries")
fig3.show()

### *FINDINGS/RESULTS*

##### Following findings have been founded from the given dataset. 
##### A) The Sunset Oasis Hotel should build Spa because its average hotel stars rating only is the highest i.e 4.3. Second option should be to build a Casino as its average hotel stars rating only is 4.2. The last option should be Tennis Court as its average hotel stars rating only is 3.9. 
##### B) The increase in rating of the Sunset Oasis Hotel by building a Spa is 0.2, by building a Tennis Court is 0.1, and by building the the Casino is -0.1. Hence, the most favourable condition is to build the spa.
##### C) Graph 1 shows that the target audience of the Sunset Oasis Hotel should be Couples. They should focus on Couples to generate revenue from their hotel. The analysis on helpful votes also supports our argument. After Couples, the hotel management should focus on the Friends group. Hotel should fulfill all the demands of couples and facilitate them in order to increase the business.
##### D) Graph 2 also shows that the target audience of the Sunset Oasis Hotel should be Couples, as the majority of guests are Couples.
##### E) Graph 3 shows that the target audience of the Sunset Oasis Hotel should belong to Taiwan, Scotland, Puerto Rico, France and Denmark as their average score rating for hotels is 5.
##### E) Graph 4 shows that the target audience of the Sunset Oasis Hotel should belong to North America as majority of guests in the given hotels belongs to North America.
##### F) Graph 5 shows that the target audience of the Sunset Oasis Hotel should belong to USA as majority of guests in the given hotels belongs to USA.
##### G) The Hotels on average have 2196 rooms and the Member years on average have 4.3 years.

### *TIPS TO INCREASE THE HOTEL RATING*
#### So how do you get lots of good reviews? By providing an outstanding guest experience, obviously. It’s an easy thing to say, but not every hotel puts it into practice. Here are some tips to help you improve your TripAdvisor hotel rating and ranking.

#### **1. Be transparent with your guests**
#### Don’t overpromise and underdeliver. Guests don’t like to be surprised – they won’t be happy to find out the 10-minute walk to the beach is actually a 15 minute trek across two busy roads and a parking lot. Ensure that all descriptions across your site are apt and accurate. If you don’t offer tea or coffee and the nearest supermarket is half an hour away, say so. People value honesty. Likewise, be sure to sing about your property’s best USPs to make up for anything that might be off-putting.

#### **2. Take complaints seriously**
#### Unhappy guests are likely to come to you first before leaving a negative review on TripAdvisor. If a customer makes a complaint, assure them that you’ll aim to fix the problem as soon as possible or provide a viable alternative. Fixing the problem quickly, then and there, can minimize the risk of the customer venting publicly about their grievance on a review site.

#### You can also let the customer know, either upon check-in or in their welcome book, that you offer an in-house feedback box or guest survey to encourage them to direct their feedback straight to you. There are few things more frustrating than feeling like you’re not being heard or taken seriously, so as long as you do this it can make a huge difference to your guest’s happiness.

#### **3. Work with your staff**
#### Your staff need to understand and be reminded of the importance of reviews for the success of your hotel. Staff should encourage reviews, especially when interacting with guests who talk to them very positively about their experience. You can nudge selected guests for a review on TripAdvisor by sending them an post-stay email asking if they enjoyed their time at your hotel, including a direct link to the hotel’s page on the review site – we recommend doing this as part of the contact-free guest journey.

#### 4. **Limit extra charges**
#### Ensure guests don’t get any nasty surprises when it’s time to settle the bill by letting them know about any extra expenditures. Nowadays, certain services such as drinking water, shuttle buses and Wi-Fi access, should all be free of charge. If they’re not, customers will be straight on TripAdvisor to complain.

#### 5. **Always respond to reviews**
#### Check your reviews constantly and aim to reply to each one promptly and with an impartial, understanding tone. Everyone gets bad reviews, that’s a given in the hospitality industry. However, replying to negative reviews isn’t so much about soothing the person who left the review, but more about allaying the fears of the many people who will read your response later. How you address the customer’s concerns will tell readers a lot about how your hotel staff and management conduct themselves in the face of dissatisfaction. Be sure to avoid responses that sound too generic – your replies should address the specific complaints detailed in the original review.

#### 6. **Take criticism on board**
#### If you see the same complaints and suggested areas for improvement cropping up over and over again, you can’t afford to ignore it, hard as it might be to admit that there’s a real problem. Listen to your guests and work proactively to avoid further complaints by improving the experiences and facilities they’re mentioning.

#### If you have a few guests complaining about the quality of the towels in their bathroom, the loss of potential bookings due to their comments is likely to outweigh the cost of investing in newer, better quality towels. If you then reply to the original comment relaying that their suggestions have been implemented, future customers will see how well you have handled the situation and may consider booking.

### *TIPS TO INCREASE THE HOTEL REVENUE*
#### **1. Offer Early Check-In and Late Checkout**

#### One of the simplest ways to increase revenue at your hotel is to charge for certain premium services instead of simply giving them away for free. Many hotel guests want the ability to check-in to their rooms early and checkout of their rooms late. A staggering number of hotels are currently honoring these requests for free. These fees can substantially increase hotel revenue and the data shows that customers are more than willing to pay anywhere from $20 to $50 for an early check-in. Some hotels are using this add-on to generate over $30k a year in fees. Imagine giving away $30k worth of services every year! 
#### On top of that, early check-in and late checkout fees are nearly 100% profit as the revenue drops straight to the bottom line results of the hotel. Not many revenue drivers can make the same impact on the P&L. 

#### **2. Promote your food and beverage options throughout the stay**
#### Modern hotel guests are used to the on-demand convenience of ecommerce options offered by companies like Amazon, DoorDash and Uber. Customers today want to be able to order what they want, when they want, and have it delivered quickly. 

#### **3. Take advantage of other upsell opportunities**

#### Data indicates that there are 4 other common upselling opportunities for hotels. If your property is not already offering these upsells to guests, you may be missing out on valuable revenue. 
#### Parking Fees: Many hotels have added parking and valet fees to their standard billing practice
#### Pet Fees: Many guests will gladly pay an additional fee to bring their pets along with them. This can add to your cleaning costs, however. 
#### Shuttle Transportation: Guests will pay for convenience. Offering a shuttle to the airport or local attractions is a great way to increase revenue. 
#### Retail or Gift Store: Hoteliers can increase revenue by offering the things their guests want to purchase in-house in a gift store or retail shop.  

#### **4. Market your hotel to appease specific guest demographics**
#### Now that “work from home” is the norm, you could advertise your hotel as the perfect “work from vacation” destination. Advertising options like this will capture the attention of a certain demographic that will be more inclined to book a room. 
#### Many hoteliers were successful earlier in the pandemic by offering promotions for hospital workers. Staying abreast of trends and events - and creating targeted promotions - can be beneficial for both your hotel and the local community. 

#### **5. Optimize your website, social media, and prices** 
#### Most travelers and tourists discover hotels and resorts through online advertising and review sites. It is imperative that you optimize your online presence to cater to the guests who are actually looking for your hotel. 
#### This means improving online reviews, increasing your presence and impact on social media, and optimizing prices to attract the right customers. Optimization does not happen overnight. It will take time to figure out your customer profile and the best ways to use content on your website to boost natural traffic.

‍